### PR TITLE
dockerTools.buildLayeredImage: Fix cross compilation

### PR DESCRIPTION
**Pull request by adisbladis**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/docker/default.nix [hunk 1]
```diff
@@ -56,6 +57,15 @@ let
...
   '';
 
+  # Map nixpkgs architecture to Docker notation
+  getArch = nixSystem: {
+    aarch64-linux = "arm64v8";
+    armv7l-linux = "arm32v7";
+    x86_64-linux = "amd64";
+    powerpc64le-linux = "ppc64le";
+    i686-linux = "i386";
+  }.${nixSystem};
```
Comment (roberth) [inline pkgs/build-support/docker/default.nix:67]:
```suggestion
  }.${nixSystem} or "Can't map Nix system ${nixSystem} to Docker architecture notation. Please check that your input and your requested build are correct or update the mapping in Nixpkgs.";
```

File: pkgs/build-support/docker/default.nix [hunk 2]
```diff
@@ -56,6 +57,15 @@ let
     done;
   '';
 
+  # Map nixpkgs architecture to Docker notation
```
Comment (roberth) [inline pkgs/build-support/docker/default.nix:60]:
```suggestion
  # Map nixpkgs architecture to Docker notation
  # Reference: https://docs.docker.com/docker-for-mac/multi-arch/
```

Unless you have a more specific link.

**Review by roberth — COMMENTED** [review]
Thanks!
Let's make sure this doesn't break again by adding a test. Could you
 - add an example of cross compilation to `pkgs/build-support/docker/examples.nix`?
 - add a test case in `nixos/tests/docker-tools.nix`? Just `docker load`ing it is already useful. If that's not possible, just reference the example in a comment that explains why we can't easily load it yet.

You may be able to `run` the cross compiled image by loading it in a separate NixOS test that runs on the intended system type. Is that something you're comfortable with?

**Comment by adisbladis** [comment]
@roberth All taken care of :)

**Review by roberth — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**